fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualWritePlatformServiceImpl.java

```diff
@@ -265,14 +265,11 @@ private void addAccrualAccounting(LoanScheduleAccrualData scheduleAccrualData, B
             BigDecimal totalAccInterest, BigDecimal feePortion, BigDecimal totalAccFee, BigDecimal penaltyPortion,
             BigDecimal totalAccPenalty, final LocalDate accruedTill) throws DataAccessException {
         AppUser user = context.authenticatedUser();
-        String transactionSql = "INSERT INTO m_loan_transaction  (loan_id,office_id,is_reversed,external_id,transaction_type_enum,transaction_date,amount,interest_portion_derived,"
-                + "fee_charges_portion_derived,penalty_charges_portion_derived, submitted_on_date, created_by, last_modified_by, created_on_utc, last_modified_on_utc) "
-                + "VALUES (?, ?, false, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
-        this.jdbcTemplate.update(transactionSql, scheduleAccrualData.getLoanId(), scheduleAccrualData.getOfficeId(),
-                externalIdFactory.create().getValue(), LoanTransactionType.ACCRUAL.getValue(), accruedTill, amount, interestPortion,
-                feePortion, penaltyPortion, DateUtils.getBusinessLocalDate(), user.getId(), user.getId(),
-                DateUtils.getOffsetDateTimeOfTenantWithMostPrecision(), DateUtils.getOffsetDateTimeOfTenantWithMostPrecision());
-        final Long transactionId = this.jdbcTemplate.queryForObject("SELECT " + sqlGenerator.lastInsertId(), Long.class); // NOSONAR
+
+        Loan loan = loanRepositoryWrapper.findOneWithNotFoundDetection(scheduleAccrualData.getLoanId());
```

Comment:
Please do not load the full loan here!
The LoanScheduleAccrualData contains the loanId and the officeId, only load the references of them please!